apps/gitness/src/pages/pull-request/sandbox-pull-request-conversation-page.tsx

```diff
@@ -0,0 +1,367 @@
+import { Spacer } from '@harnessio/canary'
+import {
+  commentCreatePullReq,
+  EnumCheckStatus,
+  EnumMergeMethod,
+  mergePullReqOp,
+  OpenapiMergePullReq,
+  reviewerDeletePullReq,
+  TypesPullReqActivity,
+  TypesPullReqReviewer,
+  useCodeownersPullReqQuery,
+  useListPullReqActivitiesQuery,
+  useReviewerListPullReqQuery
+} from '@harnessio/code-service-client'
+import {
+  SandboxLayout,
+  PullRequestCommentBox,
+  PullRequestFilters,
+  PullRequestOverview,
+  PullRequestPanel,
+  PullRequestSideBar,
+  SkeletonList
+} from '@harnessio/playground'
+import { useDateFilters } from './hooks/useDataFilters'
+import { useActivityFilters } from './hooks/useActivityFilters'
+import { useGetRepoRef } from '../../framework/hooks/useGetRepoPath'
+import {
+  capitalizeFirstLetter,
+  checkIfOutdatedSha,
+  extractInfoForCodeOwnerContent,
+  findChangeReqDecisions,
+  findWaitingDecisions,
+  generateAlphaNumericHash,
+  processReviewDecision
+} from './utils'
+import { useParams } from 'react-router-dom'
+import { PathParams } from '../../RouteDefinitions'
+import { usePullRequestData } from './context/pull-request-data-provider'
+import { isEmpty } from 'lodash-es'
+import { CodeOwnerReqDecision } from '../../types'
+import { useAppContext } from '../../framework/context/AppContext'
+
+export default function SandboxPullRequestConversationPage() {
+  const {
+    refetchActivities,
+    refetchPullReq,
+    pullReqMetadata,
+    pullReqChecksDecision,
+    prPanelData,
+    setRuleViolationArr,
+    loading: prLoading
+  } = usePullRequestData()
+  const { currentUser: currentUserData } = useAppContext()
+  const [checkboxBypass, setCheckboxBypass] = useState(false)
+  const { spaceId, repoId } = useParams<PathParams>()
+
+  const repoRef = useGetRepoRef()
+  const { pullRequestId } = useParams<PathParams>()
+  const prId = (pullRequestId && Number(pullRequestId)) || -1
+  // const [loadState, setLoadState] = useState('data-loaded')
+  const dateFilters = useDateFilters()
+  const [dateOrderSort, setDateOrderSort] = useState<{ label: string; value: string }>(dateFilters[0])
+  const activityFilters = useActivityFilters()
+  const [activityFilter, setActivityFilter] = useState<{ label: string; value: string }>(activityFilters[0])
+  const { data: { body: reviewers } = {}, refetch: refetchReviewers } = useReviewerListPullReqQuery({
+    repo_ref: repoRef,
+    pullreq_number: prId
+  })
+  const { data: { body: codeOwners } = {}, refetch: refetchCodeOwners } = useCodeownersPullReqQuery({
+    repo_ref: repoRef,
+    pullreq_number: prId
+  })
+  const { data: { body: activityData } = {} } = useListPullReqActivitiesQuery({
+    repo_ref: repoRef,
+    pullreq_number: prId,
+    queryParams: {}
+  })
+  const [changesLoading, setChangesLoading] = useState(true)
+
+  const [activities, setActivities] = useState(activityData)
+  const approvedEvaluations = reviewers?.filter(evaluation => evaluation.review_decision === 'approved')
```

Comment:
Could filters be memoized?

Reply:
@cjlee01 @vardanbansal-harness let's meet sometime this week to move everything out of state for this page, and into a more suitable storage (context/reducer)

Reply:
sure , can we sync tomoro @abhinavrastogi-harness  @vardanbansal-harness